bot/decorators.py

```diff
@@ -145,6 +152,8 @@ async def wrapper(*args, **kwargs) -> t.Any:
                     return await func(*args, **kwargs)
             else:
                 log.info(f"{name}: aborted because resource {namespace!r}:{id_!r} is locked")
+                if raise_error:
+                    raise LockedResourceError(str(namespace), id_)
```

Comment:
I like this. Good way to signal to the caller/error handler what's up and allow them to handle it.